proxy/src/dns.rs

```diff
@@ -72,20 +85,44 @@ impl Config {
 
 impl Resolver {
     pub fn new(config: Config, executor: &Handle) -> Self {
-        Resolver(ns_dns_tokio::DnsResolver::new_from_resolver(
-            domain::resolv::Resolver::from_conf(executor, config.0),
-        ))
+        Resolver {
+            resolver: ns_dns_tokio::DnsResolver::new_from_resolver(
+                domain::resolv::Resolver::from_conf(executor, config.0)),
+            executor: executor.clone()
+        }
     }
 
-    pub fn resolve_host(&self, host: &transport::Host) -> IpAddrFuture {
+    pub fn resolve_one_ip(&self, host: &transport::Host) -> IpAddrFuture {
         match *host {
             transport::Host::DnsName(ref name) => {
-                trace!("resolve {}", name);
-                IpAddrFuture::DNS(self.0.resolve_host(&name.0))
+                trace!("resolve_one_ip {}", name);
+                IpAddrFuture::DNS(self.resolver.resolve_host(&name.0))
             }
             transport::Host::Ip(addr) => IpAddrFuture::Fixed(addr),
         }
     }
+
+    pub fn resolve_all_ips(&self, delay: Duration, host: &Name) -> IpAddrListFuture {
```

Comment:
I don't have a strong feeling on this one way or another, but did you consider making this return an `IpAddrListStream` instead (i.e. pushing the TTL/looping down into stream impl). The main advantage of this, I think, would be that the `poll_dns` implementation in `control::discovery` could be moved into the `dns` module, which (I think) means that discovery won't have to change to support TTLs, etc...

Reply:
to be clear: it would be fine to do this in a followup; i'm just curious if you avoided pushing the looping down for a specific reason.

Reply:
> to be clear: it would be fine to do this in a followup; i'm just curious if you avoided pushing the looping down for a specific reason.

Converting this to return a Stream is a good idea. I was meaning to follow up with a change that did that, but I am running short on time. If it is OK to do that as a follow-up, let's do so.